lib/sycamore/sycamore/tests/unit/test_docset.py

```diff
@@ -267,6 +267,23 @@ def num_as(self, doc: Document) -> int:
 
         return count
 
+    def _test_limit(self, exec_mode):
+        texts = [self.random_string(min_size=20, max_size=100) for _ in range(10)]
+        docs = [Document(text_representation=t, doc_id=i, properties={}) for i, t in enumerate(texts)]
+
+        context = sycamore.init(exec_mode=exec_mode)
+
+        for i in [0, 4, 10]:
+            docset = context.read.document(docs).limit(i)
+            assert docset.count() == i
+            assert len(set(d.doc_id for d in docset.take_all())) == i
+
+    def test_limit_local(self):
+        self._test_limit(sycamore.EXEC_LOCAL)
+
+    def test_limit_ray(self):
```

Comment:
Can you move this into the integration tests? I'm hoping to make the unit tests really fast. 
Here's two patterns I've been using:
https://github.com/search?q=repo%3Aaryn-ai%2Fsycamore+path%3A%2F%5Elib%5C%2Fsycamore%5C%2Fsycamore%5C%2Ftests%5C%2Fintegration%5C%2F%2F+%2Fexec%5CS*ray%2F&type=code

I think inherit and get the execmode from self would work well here; but either choice works.